buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy

```diff
@@ -49,25 +55,38 @@ public class NoticeTask extends DefaultTask {
         dependencies.add(project)
     }
 
+    /** Add notices from an extra directory. */
+    public void extraLicensesDir(File extraLicensesDir) {
+        extraLicensesDirs.add(extraLicensesDir)
+    }
+
     @TaskAction
     public void generateNotice() {
         StringBuilder output = new StringBuilder()
         output.append(inputFile.getText('UTF-8'))
         output.append('\n\n')
-        Set<String> seen = new HashSet<>()
+        Map<String, File> seen = new TreeMap<>()
         for (Project dep : dependencies) {
```

Comment:
We can change this to iterate over the directories. We don't need to have dependencies at all here (I think) since DependencyLicensesTask checks there are not extra bogus notice/license files not associated with our dependencies in the licenses dir.